frontend/src/bundles/common/components/dropdown/dropdown.tsx

```diff
@@ -0,0 +1,83 @@
+import { useCallback, useState } from 'react';
+import {
+    type Control,
+    type FieldPath,
+    type FieldValues,
+} from 'react-hook-form';
+
+import { useFormController } from '../../hooks/hooks';
+import { Option } from './option/option';
+import styles from './styles.module.scss';
+
+type Properties<T extends FieldValues> = {
+    label: string;
+    control: Control<T, null>;
+    name: FieldPath<T>;
+    defaultOption: string;
+    options: {
+        value: string;
+        label: string;
+    }[];
+};
+
+const Dropdown = <T extends FieldValues>({
+    control,
+    name,
+    label,
+    defaultOption,
+    options,
+}: Properties<T>): JSX.Element => {
+    const { field } = useFormController({
```

Comment:
This component knows too much about the environment it assumed to be used. Avoid using `react-hook-form` and similar packages inside the component